api4/post.go

```diff
@@ -733,7 +733,7 @@ func patchPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	auditRec.Success()
 	auditRec.AddMeta("patch", patchedPost)
 
-	if err := json.NewEncoder(w).Encode(patchedPost); err != nil {
+	if err := patchedPost.EncodeJSON(w); err != nil {
```

Comment:
Instead of using a separate method like `EncodeJSON` it might be cleaner to implement the [`json.Marshaler`](https://pkg.go.dev/encoding/json#Marshaler) interface and keep using `Encode`.

Reply:
I see your concerns about overwriting the `MarshalJSON`. Would it make sense to rename `EncodeJSON` to something that makes it clear that this methods should be used for communication with clients?